packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,249 @@ class HomeScreen extends StatefulWidget {
+        imageUrl = '';
+      });
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.error('Delete error - ${e.message}');
+    }
+  }
+
+  // get the url of a file in the S3 bucket
+  Future<String> getUrl({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      final result = await Amplify.Storage.getUrl(
+        key: key,
+        options: StorageGetUrlOptions(
+          accessLevel: accessLevel,
+          pluginOptions: const S3GetUrlPluginOptions(
+            validateObjectExistence: true,
+            expiresIn: Duration(minutes: 1),
+          ),
+        ),
+      ).result;
+      setState(() {
+        imageUrl = result.url.toString();
+      });
+      return result.url.toString();
+    } on StorageException catch (e) {
+      _logger.error('Get URL error - ${e.message}');
+      rethrow;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Amplify Storage Example'),
       ),
-      body: const Center(
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text('Examples are coming...'),
-          ],
-        ),
+      body: Stack(
+        children: [
+          Center(
+            child: Padding(
+              padding: const EdgeInsets.all(20),
+              child: ListView(
+                children: [
+                  for (final item in list)
+                    ListTile(
+                      onTap: () => {
+                        getUrl(
+                          key: item.key,
+                          accessLevel: StorageAccessLevel.guest,
+                        )
+                      },
+                      title: Text(item.key),
+                      trailing: Row(
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          IconButton(
+                            icon: const Icon(Icons.delete),
+                            onPressed: () {
+                              removeFile(
+                                key: item.key,
+                                accessLevel: StorageAccessLevel.guest,
+                              );
+                            },
+                            color: Colors.red,
+                          ),
+                        ],
+                      ),
+                      leading: IconButton(
+                        icon: const Icon(Icons.download),
+                        onPressed: () {
+                          if (defaultTargetPlatform == TargetPlatform.iOS ||
+                              defaultTargetPlatform == TargetPlatform.android) {
+                            downloadFileMobile(item.key);
+                          } else {
+                            downloadFileWeb(item.key);
+                          }
+                        },
```

Comment:
The "mobile" version should work on desktop platforms as well:
```suggestion
                        onPressed: () => zIsWeb
                            ? downloadFileWeb(item.key)
                            : downloadFileMobile(item.key),
```